sdk/monitor/Azure.Monitor.Ingestion/tests/IngestionClientTest.cs

```diff
@@ -0,0 +1,227 @@
+            var currentTime = DateTimeOffset.UtcNow.ToString("O");
+
+            BinaryData data = BinaryData.FromObjectAsJson(
+                // Use an anonymous type to create the payload
+                new[] {
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer1",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user1",
+                            TimeZone = "Pacific Time",
+                            Level = 4,
+                            CounterName = "AppMetric1",
+                            CounterValue = 15.3
+                        }
+                    },
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer2",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user2",
+                            TimeZone = "Central Time",
+                            Level = 3,
+                            CounterName = "AppMetric1",
+                            CounterValue = 23.5
+                        }
+                    },
+                });
+
+            LogsIngestionClient client = new(
+                new Uri(dcrEndpoint),
+                clientSecretCrendential);
+
+            // Make the request
+            Response response = client.Upload(dcrImmutableId, "Custom-MyTableRawData", RequestContent.Create(data)); //takes StreamName not tablename
+
+            // Check the response
+            Assert.AreEqual(204, response.Status);
+            Assert.AreEqual("", response.Content.ToString());
+
+            LogsQueryClient logsQueryClient = new LogsQueryClient(clientSecretCrendential);
+            var batch = new LogsBatchQuery();
+            string workspaceId = "22101c38-d67f-4ded-994e-67f093aff9f4";
+            string countQueryId = batch.AddWorkspaceQuery(
+                workspaceId,
+                "MyTable_CL | count;",
+                new QueryTimeRange(TimeSpan.FromDays(1)));
+
+            Response<LogsBatchQueryResultCollection> responseLogsQuery = await logsQueryClient.QueryBatchAsync(batch);
+
+            // Check the Azure.Monitor.Query Respose
+            Assert.AreEqual(200, responseLogsQuery.GetRawResponse().Status);
+            Assert.AreEqual("OK", responseLogsQuery.GetRawResponse().ReasonPhrase);
+            Assert.IsTrue(responseLogsQuery.Value.GetResult<int>(countQueryId).Single() >= 2);
+        }
+
+        [Test]
+        public async Task ValidInputFromStreamAsync() //need Workspace id
+        {
+            ClientSecretCredential clientSecretCrendential = new ClientSecretCredential("<tenantId>", "<cliendId>", "<clientSecret>");
+            var dcrImmutableId = "<dcrImmutableId>";
+            var dcrEndpoint = "https://nibhati-dce-ku6s.westus2-1.ingest.monitor.azure.com";
```

Comment:
Given how common all the client creation code is, most folks extract that out to a helper method.  Imagine something like `IngestionClient client = CreateTestClient();` that uses the `IngestionClientTestEnvironment` to pull all the settings and correctly instantiate things.  You'll save a few lines per test and have to update fewer things when the config changes.  Ditto for the LogsQueryClient too.